docs/core/diagnostics/observability-with-otel.md

```diff
@@ -0,0 +1,426 @@
+
+In a later stage we will feed this data into Jaeger to visualize the distributed traces.
+
+### 7. Install and use Prometheus
+
+The metrics data that is exposed in Prometheus format is a point in time snapshot of the processes metrics. Each time a request is made to the endpoint, it will give the current values. While current values are interesting, they become most valuable when compared to previous values, so you can see trends, and if they are anomylous. Commonly services have spikes based on the time of day, or world events such a shopping spree on black friday. Only by comparing the values can you detect if they are abnormal, or if a metric is getting progressively worse.
+
+The process doesn't store and aggregate any history of the metrics. Adding that capability to the process could be resource intensive, and in a distributed system we commonly have multiple instances or each node, so we want to be able to collect the metrics from all of them, and then aggregate and compare with historical values. This is where Prometheus comes in, you configure it with the metric endpoints and it will periodically scrape the values and store then in a time series database. They can then be analyzed and processed as needed.
+
+Download Prometheus for your platform from [https://prometheus.io/download/](https://prometheus.io/download/) and extract the contents of the download.
+
+Look at the top of the output of our running server to get the port number for the http endpoint. For example:
+
+``` shell
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: https://localhost:7275
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5212
+```
+
+Modify the Prometheus yaml file to specify the port for our http scraping endpoint, and set a lower scraping interval eg:
+
+``` yaml
+  scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: "prometheus"
+
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+
+    static_configs:
+      - targets: ["localhost:5212"]
+      - scrape_interval: 1s # poll very quickly for a more responsive demo
+```
+
+Start Prometheus, and look in the output for the port its running on, typically 9090
+
+``` shell
+>prometheus.exe
+...
+ts=2023-06-16T05:29:02.789Z caller=web.go:562 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
+```
+
+Open this URL in your browser. In the Prometheus UI you should now be able to query for your metrics, you use the highlighted button, it will bring up the metrics explorer which will show all the metrics available.
+
+![Prometheus Metrics Explorer](./media/prometheus-metrics-explorer.png)
+
+Select the `greetings_count` metric to see a graph of values.
+
+![Graph of greetings_count](./media/prometheus-graph.png)
+
+### 8. Install Grafana and create a dashboard
```

Comment:
We also have Azure Managed Grafana.
https://learn.microsoft.com/azure/managed-grafana/overview